tower-grpc-build/src/server.rs

```diff
@@ -411,32 +412,38 @@ macro_rules! try_ready {
 
                 is_end_stream_block
                     .line(&format!(
-                        "Ok({}(ref v)) => v.is_end_stream(),",
+                        "{}(ref v) => v.is_end_stream(),",
                         &upper_name
                     ));
 
                 poll_data_block
                     .line(&format!(
-                        "Ok({}(ref mut v)) => v.poll_data(),",
+                        "{}(ref mut v) => v.poll_data(),",
                          &upper_name
                     ));
 
                 poll_metadata_block
                     .line(&format!(
-                        "Ok({}(ref mut v)) => v.poll_metadata(),",
+                        "{}(ref mut v) => v.poll_metadata(),",
                          &upper_name
                     ));
             }
 
-            is_end_stream_block.line("Err(_) => true,");
-            poll_data_block.line("Err(_) => Ok(None.into()),");
-
-        let mut poll_metadata_catch_all = codegen::Block::new("Err(ref status) =>");
-            poll_metadata_catch_all
-            .line("status.to_header_map().map(Some).map(Into::into)")
-            ;
-
-            poll_metadata_block.push_block(poll_metadata_catch_all);
+            is_end_stream_block
+                .line(&format!(
+                    "{}(_) => true,",
```

Comment:
I'm a bit confused. The PR description stated that the problem is that the unimplemented variant's `is_end_stream()` fn would return `true`, so trailers were not sent. However, as far as I can tell, this behavior is still present.

Reply:
Instead of just changing the `is_end_stream` function, so that it would be returned in the trailers, this changes it so that the status is returned in the headers.